src/Farmer/Types.fs

```diff
@@ -199,6 +199,19 @@ type LinkedResource =
         | Unmanaged resId -> resId
     member this.Name = this.ResourceId.Name
 
+type SslState = 
+    | Sni of thumbprint:ArmExpression
```

Comment:
Ah I see some of the problem here. We probably need two types for this, one we use at the arm layer as you had before and one which doesn't take a thumbprint which we use here. So that, for AppManagedCertificate, we can generate the certificate and thumbprint expression